examples/multisig_plain/lib.rs

```diff
@@ -0,0 +1,837 @@
+//! the usual submit, confirm, execute cycle as any other transaction that should be
+//! called by the wallet. For example to add an owner you would submit a transaction
+//! that calls the wallets own `add_owner` message through `submit_transaction`.
+//!
+//! ### Owner Management
+//!
+//! The messages `add_owner`, `remove_owner`, and `replace_owner` can be used to manage
+//! the owner set after instantiation.
+//!
+//! ### Changing the Requirement
+//!
+//! `change_requirement` can be used to tighten or relax the `requirement` of how many
+//! owner signatures are needed to execute a `Transaction`.
+//!
+//! ### Transaction Management
+//!
+//! `submit_transaction`, `cancel_transaction`, `confirm_transaction`,
+//! `revoke_confirmation` and `execute_transaction` are the bread and butter messages
+//! of this contract. Use them to dispatch arbitrary messages to other contracts
+//! with the wallet as a sender.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_lang as ink;
+
+#[ink::contract(version = "0.1.0", env = ink_core::env::DefaultEnvTypes)]
+mod multisig_plain {
+    use ink_core::{
+        env,
+        storage,
+    };
+    use ink_prelude::vec::Vec;
+    use scale::Output;
+
+    /// Tune this to your liking but be wary that many owners will not perform well.
+    const MAX_OWNERS: u32 = 50;
+
+    type TransactionId = u32;
+    const WRONG_TRANSACTION_ID: &str =
+        "The user specified an invalid transaction id. Abort.";
+
+    /// A wrapper that allows us to pass untyped parameters as blob to a `CallBuilder`
+    struct CallInput<'a>(&'a [u8]);
+
+    impl<'a> scale::Encode for CallInput<'a> {
+        fn encode_to<T: Output>(&self, dest: &mut T) {
+            dest.write(self.0);
+        }
+    }
+
+    /// A Transaction is what every `owner` can submit for confirmation by other owners.
+    /// If enough owners agree it will be executed by the contract.
+    #[derive(scale::Encode, scale::Decode, storage::Flush)]
+    #[cfg_attr(feature = "ink-generate-abi", derive(type_metadata::Metadata))]
+    #[cfg_attr(feature = "std", derive(Debug, PartialEq, Eq))]
+    pub struct Transaction {
+        /// The AccountId of the contract that is called in this transaction.
+        callee: AccountId,
+        /// The raw selector which is the function name of the `callee`that is called.
+        selector: [u8; 4],
+        /// The raw parameters that are passed to the called function.
+        input: Vec<u8>,
+        /// The amount of chain balance that is transferred to the callee.
+        transferred_value: Balance,
+        /// Gas limit for the transation.
+        gas_limit: u64,
+    }
+
+    #[ink(storage)]
+    struct MultisigPlain {
+        /// Every entry in this map represents the confirmation of an owner for a
+        /// transaction. This is effecively a set rather than a map.
+        confirmations: storage::BTreeMap<(TransactionId, AccountId), ()>,
```

Comment:
I decided against that when implementing this contract for the following reason: You need stable owner indices for that. If you do enough owner removals additions and replacements you finally run out of indices and your contract is broken because you ran out of indices.

Maybe we can live with that because 64 ids is plenty but I wanted to opt for the safe approach in this example.